spec/features/cdr/cdr_history/cdr_filter_spec.rb

```diff
@@ -0,0 +1,72 @@
+RSpec.describe 'Cdrs index page filtering', js: true do
+  subject do
+    within_filters { click_submit('Filter') }
+  end
+
+  include_context :login_as_admin
+
+  before do
+    Cdr::Cdr.delete_all
+    visit cdrs_path
+
+    within_filters do
+      select_by_value tagged_filter_value, from: 'Tagged'
+    end
+  end
+
+  let!(:cdrs) { create_list(:cdr, 2) }
+  let!(:tag) { create :routing_tag }
+  let!(:cdr_tagged) { create :cdr, routing_tag_ids: [tag.id] }
+  let(:tagged_filter_value) { nil }
+
+  context 'with filter by tagged' do
+    let(:tagged_filter_value) { true }
+
+    it 'shows filtered records with routing_tags' do
+      subject
+      expect(page).to have_table
+      expect(page).to have_table_row count: 1
+      expect(page).to have_table_cell column: 'Id', text: cdr_tagged.id
+      expect(page).to have_select :Tagged, selected: 'Yes'
+    end
+  end
+
+  context 'with filter by not tagged' do
+    let(:tagged_filter_value) { false }
+
+    it 'shows filtered records without routing_tags' do
+      subject
+      expect(page).to have_table
+      expect(page).to have_table_row count: 2
+      expect(page).to have_table_cell column: 'Id', text: cdrs.first.id
+      expect(page).to have_select :Tagged, selected: 'No'
+    end
+  end
+
+  context 'with filter by any tagged' do
+    it 'shows filtered records without routing_tags' do
+      subject
+      expect(page).to have_table
+      expect(page).to have_table_row count: Cdr::Cdr.count
+      expect(page).to have_table_cell column: 'Id', text: cdrs.first.id
+      expect(page).to have_table_cell column: 'Id', text: cdr_tagged.id
+      expect(page).to have_select :Tagged, selected: 'Any'
+    end
+  end
+
+  context 'with filter by routing tag ids contains' do
+    let(:tagged_filter_value) { tag.name }
+
+    it 'shows filtered records with contains ids' do
+      within_filters do
+        fill_in_chosen 'Routing Tag IDs Contains', with: tagged_filter_value
+      end
+
+      subject
+      expect(page).to have_table
+      expect(page).to have_table_row count: 1
+      expect(page).to have_table_cell column: 'Id', text: cdr_tagged.id
+      expect(page).to have_select 'q_routing_tag_ids_array_contains', selected: tagged_filter_value
```

Comment:
we should not use `have_select` here because in UI we don't show ordinary select - it being hidden by chosen.js which shows custom built element. So this test can be false positive - appropriate value can be selected in `<select>` but chosen container will be empty.

So you need to replace expectation with following

```ruby
expect(page).to have_field_chosen('Routing Tag IDs Contains', with: tagged_filter_value)
```

and add following matcher to `Helpers::Chosen`

```ruby
  # Checks chosen select presence on page
  # @param label [String] field label.
  # @param options [Hash]
  #   :with [String, nil] check selected option text whether not nil,
  #   :disabled [Boolean, nil] default false,
  #   :exact [Boolean] match :with by exact text (default true),
  #   for other options @see #have_selector.
  def have_field_chosen(label, options = {})
    with = options.delete(:with)
    disabled = options.delete(:disabled)
    exact = options.delete(:exact)
    exact = true if exact.nil?
    warn 'empty :with will be ignored because :exact is false' if with.blank? && !exact
    options[exact ? :exact_text : :text] = with unless with.nil?
    selector = chosen_container_selector(label, disabled: disabled)
    have_selector(selector, options)
  end
```

Reply:
same for `Tagged`